apps/client/src/sttr_client/models/checker.js

```diff
@@ -50,6 +50,66 @@ class Checker {
     return this._stack;
   }
 
+  /**
+   * Returns a list of questionIds with the given answers.
+   * Useful to store in React.Context or in sessionStorage
+   * @returns {({string: boolean|string|number|[string]})}  - a list of answers
+   */
+  getQuestionAnswers() {
+    return this.stack
+      .concat(this._getUpcomingQuestions()) // Merge the stack with upcoming questions to get all questions
+      .reduce((acc, question) => {
+        acc[question.id] = question.answer;
+        return acc;
+      }, {});
+  }
+
+  /**
+   * Loops through all questions and answers these questions based on provided `answers`
+   * Useful when you have stored the user answers and want to continue a particular session
+   * @returns {void} -
+   */
+  setQuestionAnswers(answers) {
+    if (!isObject(answers)) {
+      throw Error("Answers must be of type object");
+    }
+
+    // Load the first question
+    if (!this._last) {
+      this.next();
+    }
+
+    let counter = 0;
+    let done = false;
+    let prevId = null;
+
+    // Loop through `sttr-checker` and answer all questions provided by `answers`
+    while (done === false) {
+      const questionAnswer = answers[this._last.id];
+
+      // The checker is completed when `questionAnswer` is undefined
+      // or if the same question is handled again (to allow `Contact` conclusions)
+      if (questionAnswer === undefined || prevId === this._last.id) {
+        done = true;
+      } else {
+        // Answer the question and proceed
+        prevId = this._last.id;
+        this._last.setAnswer(questionAnswer);
+        this.next();
+      }
+
+      // Prevent infinite loop, otherwise the browser will explode
+      // @TODO: Add this to monitoring in Sentry
+      counter++;
+      if (counter > 1000) {
+        console.error(
+          "Infinite loop detected and prevented in setQuestionAnswers()"
+        );
+        done = true;
+      }
+    }
+  }
+
```

Comment:
What?

Reply:
I had problems with infinite loops. After I added `|| prevId === this._last.id` it seems to have been solved. But I still wanted to keep this as a backup because while (done === false) { is very dangerous.

Remove this fallback?

Reply:
Yes, let's remove it.

Reply:
Removed (-: